lib/plugins/emit/index.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+const fdk = require('@serverless/fdk');
+const path = require('path');
+const stdin = require('get-stdin');
+const userStats = require('../../utils/userStats');
+
+class Emit {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options || {};
+    this.data = null;
+
+    this.commands = {
+      emit: {
+        usage: 'Emits an event to a running Event Gateway',
+        lifecycleEvents: ['emit'],
+        options: {
+          name: {
+            usage: 'Event name',
+            required: true,
+            shortcut: 'n',
+          },
+          path: {
+            usage: 'Path to JSON or YAML file holding input data',
+            shortcut: 'p',
+          },
+          data: {
+            usage: 'input data',
+            shortcut: 'd',
+          },
+          url: {
```

Comment:
It's localhost for now, but if you want to emit to a hosted gateway it will be a different url. We use localhost:4000 as default, but this is a feature already optimised to be future-proof. I suggest to keep url